### DISCUSSION TITLE
Deploying to AWS with NextJS API Routes

### DISCUSSION THREAD

**Discussion by standingdreams** [vercel/next.js#14362] [Help]
I was hosting my app on Vercel but have decided to find another solution. I have yet to find a proper solution to deploy to any AWS service if you are using API Routes. I've been searching for a while. 

I've attempting to deploy my app to AWS using Serverless. However, there is currently some issues with serverless-nextjs and `getInitialProps` or `getServerSideProps` that does not allow for me to deploy properly. I've attempt Amplify and have not been successful. I know that it could be due to my lack of knowledge.

Is Lightsail hosting on Vercel the only solution we have when using API Routes? Has anyone found any success anywhere else? If so, mind sharing details?

**Comment by rafinskipg** [comment]
I would like to see what people know about this.

**Reply by standingdreams** [reply]
RIGHT! Because it seems like no one is sharing this information. I've searched high and low. All resources point to Serverless or only discuss static sites. Nothing about the API Routes.

**Reply by EliasTouil** [reply]
> I would like to see what people know about this.

Yes please

**Reply by caesarbell** [reply]
I stumble across [this](https://egghead.io/lessons/javascript-how-to-deploy-a-nextjs-app-to-aws-with-api-routes-and-dynamic-routes) during my research for a similar solution. Deploying a NextJS application to a non-Vercel hosting service (i.e AWS) with API routes enable
